internal/infrastructure/kubernetes/proxy/resource_provider.go

```diff
@@ -146,13 +146,24 @@ func (r *ResourceRender) ServiceAccount() (*corev1.ServiceAccount, error) {
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace:       r.Namespace(),
 			Name:            r.Name(),
-			Labels:          labels,
+			Labels:          saLabels,
 			Annotations:     r.infra.GetProxyMetadata().Annotations,
 			OwnerReferences: r.OwnerReferences(),
 		},
 	}, nil
 }
 
+// envoyLabels returns the labels, including extraLabels, used for Envoy resources.
+func (r *ResourceRender) envoyLabels(extraLabels map[string]string) map[string]string {
```

Comment:
unrelated to this PR, but this logic allows a user to override EG labels, right ?

Reply:
Theoretically, it's possible.